src/pybind/mgr/cephadm/serve.py

```diff
@@ -111,6 +113,35 @@ def serve(self) -> None:
             self.log.debug("serve loop wake")
         self.log.debug("serve exit")
 
+    def _check_certificates(self) -> None:
+        cert = self.mgr.get_store('grafana_crt')
+        key = self.mgr.get_store('grafana_key')
+        if not (cert and key):
+            self.mgr.remove_health_warning('GRAFANA_CERT_ERROR')
+            return
+
+        (org, cn) = (None, None)
+        try:
+            (org, cn) = get_cert_issuer_info(cert)
+            verify_tls(cert, key)
+            self.mgr.remove_health_warning('GRAFANA_CERT_ERROR')
+        except ServerConfigException as e:
+            err_msg = """
+            Detected invalid grafana certificates. Set mgr/cephadm/grafana_crt
+            and mgr/cephadm/grafana_key to valid certificates or reset their value
+            to an empty string in case you want cephadm to generate self-signed Grafana
+            certificates.
+
+            Once done, run the following command to reconfig the daemon:
+
+               > ceph orch daemon reconfig <grafana-daemon>
+
+            """
+            self.log.error(f'Invalid grafana certificate: {e}')
+            self.mgr.set_health_warning('GRAFANA_CERT_ERROR',
+                                        f'Invalid grafana certificate: {e}',
+                                        1, [err_msg])
```

Comment:
I think we can also remove the health warning if we didn't hit an exception when verifying the cert/key.

Reply:
Not sure if I get you here but we are already removing the health warning in case of success by calling:

`            self.mgr.remove_health_warning('GRAFANA_CERT_ERROR')
`

Reply:
ACK, missed that somehow